src/js/tools/drawing/ShiftIndex.js

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable indent */
+/**
+ * @provide pskl.tools.drawing.ShiftIndex
+ *
+ * @require Constants
+ * @require pskl.utils
+ */
+(function() {
+	var ns = $.namespace('pskl.tools.drawing');
+
+	ns.ShiftIndex = function() {
+		this.superclass.constructor.call(this);
+
+		this.toolId = 'tool-shift-index';
+		this.helpText = 'ShiftIndex';
+		this.shortcut = pskl.service.keyboard.Shortcuts.TOOL.SHIFT_INDEX;
+
+		this.tooltipDescriptors = [{ key: 'ctrl', description: 'Shift Index backwards' }];
```

Comment:
Promotion describes is as follows

 Home Pro MotionDocumentation
Documentation
Menu Mode
This menu is a list of the different drawing modes. Commonly whenever you draw with the foreground color (drawing with left mouse button) the mode you have chosen here will be used. This mode also affects drawing primitives like lines, rectangles, circles...
Whenever possible the right mouse button uses an inverse operation like darken if the mode is Brighten.
Some paint modes do color calculations. Because there are only 256 colors that can be used the quality of the result may be poor if your palette does not contain proper entries. The more colors of different kind your color table contains the better the results will be.

Some modes can be scaled, e.g. setting up a strength value. If this is the case then a corresponding edit field appears at the label of the currently selected paint mode.

The scale can also be applied with pen pressure.

Besides selecting a paint mode from this menu you can use the shortcuts as displayed for every menu entry. Hit the same key again to switch back to the previously used mode.

Paint
Paints with current brush as it is. That means the brush is simply stamped down.

Erase
Uses the transparent color for painting if there is no alpha plane. If alpha is enabled then the left mouse button reduces the opacity by the given value where the right mouse button draws fully transparent. In detail it works like this:
Project has simple or no transparency
  any mouse button= transparent color if layer is transparent or project background color if it is not
Project has alpha transparency   left mouse button= decrease alpha by selected opacity
  right mouse button= set zero alpha

Selection
Instead of modifying pixels this mode will modify the selection. This way you can apply the drawing selection easily using any drawing tool. The left mouse button includes and the right excludes for selection.

Halftone
Uses the currently selected halftone pattern for drawing. It will use the selected foreground and background color with the pattern.

Brightness
Brightens image colors with left mouse button and darkens with the right. You can set the strength of brightening in the Settings.

Tint
Tints image colors with brush colors. That means the brightness of each image pixel is taken and the color from the corresponding brush pixel. Both are mixed together and the result is a pixel with the brush pixel color and the image pixel brightness.

Inc/Dec Hue
Takes the Hue (color model HSV) of the image pixels and rotates it (0..360°). You can set the strength of this mode in the Settings

Inc/Dec Saturation
Increments/decrements the Saturation(color model HSV) of the image pixel colors. You can set the strength and type of this mode in the Settings

Invert
Simply inverts the image colors using the RGB-color model. A Red value of f.e. 40 will then be 215 (=255-40). The mode will try to find the best matching inverse color in the palette.

Difference
Uses difference values between image and brush colors for painting (RGB color model).

Translucent
Mixes image and brush colors. You can define a proportional weighting for this mode. That means the result color is calculated by x% of the image color and y% of the brush color and x+y=100%. See the paint mode Settings below.

Color Cycle
Uses the colors of the current gradient one after another. If the last gradient color is passed the first one is used again...

Gray
Reduces the image colors to gray levels.

Rub Through
When drawing with this mode the image that resides in the "Spare Frame" (see descriptions for menu Frame) becomes visible. You "rub through" to the spare frame.

Shade
Shifts the image colors which are part of the current gradient. So for each color the next or previous color within the gradient will be used depending on left or right click. This way you can shade objects to emphasize their structure, to create different color effects or to smooth edges.
https://www.cosmigo.com/promotion/docs/onlinehelp/MenuMode.htm

Reply:
how about

	```
this.tooltipDescriptors = [
			{ 
				description: 'Shifts the colors which are a part of the current palette. ' + 
					'For each pixel color it touches, the next or previous color within the palette will be used. ' + 
					'Use the primary and secondary colors in the palette to set ' +
					'cell shade range boundaries. For example if your palette has: ' +
					'- light red, dark red, light blue, dark blue - ' +
					'you would want to set the primary colour to light red and secondary to dark red. ' +
					'This will prevent the brush from cycling red to blue'
			},
			{ key: 'ctrl', description: 'Shift Index backwards' },
		];
```